lib/common/socket/evloop.c.h

```diff
@@ -535,6 +535,37 @@ static void run_pending(h2o_evloop_t *loop)
     }
 }
 
+void h2o_evloop_destroy(h2o_evloop_t *loop)
+{
+    struct st_h2o_evloop_socket_t *sock;
+    //dispose all socket
+    while (loop->_pending_as_server != NULL || loop->_pending_as_client != NULL) {
```

Comment:
Ok, i will fix this. I am just studying your excellent program and now not understand every thing, happening in your code, because of this sometime i am just copy paste. Thank you for so detailed explanation.